site/content/guides/openapi-swagger.md

```diff
@@ -0,0 +1,230 @@
+      }
+    }
+
+    // set query parameters
+    if (httpMethod.toUpperCase() === 'GET' && this.cache === false) {
+        queryParams['_'] = new Date().getTime();
+    }
+    request.query(this.normalizeParams(queryParams));
+
+  ...
+
+```
+
+Depending on how much is described in a given API spec, it is possible to move even further and generate tests, mock servers and more. See [openapi.tools](https://openapi.tools) for an up-to-date list of tools belonging to the OpenAPI ecosystem.
+
+## Generated monitoring checks with Checkly
+
+[API monitoring](/guides/api-monitoring) is key to ensure that our endpoints are returning the correct results in an acceptable timeframe.  If we can generate APIs from a OAS description file, we can also generate monitoring checks for them. Checkly allows us to import an OpenAPI spec file and automatically creates one or more checks depending on the amount of information available. All it takes is a couple of clicks.
+
+When creating an API check, select the `import from Swagger / OpenAPI` button.
+
+{{< figure src="/guides/images/guides-checkly-openapi-check.png" alt="screenshot of checkly check api creation openapi import button" >}}
+
+That will bring up the import dialog. This is where we give Checkly the URL to the Swagger/OpenAPI specification file.
+
+{{< figure src="/guides/images/guides-checkly-openapi-dialog.png" alt="screenshot of checkly check api creation openapi import dialog" >}}
+
+Before anything is created on our behalf, Checkly will show us a summary of all the checks that will be imported. We can choose to import them all defined checks or just a subset.
+
+Notice that we can also select whether to import headers, query params and other settings or just to ignore them for now.
+
+{{< figure src="/guides/images/guides-checkly-openapi-import.png" alt="screenshot of checkly bulk openapi import" >}}
+
+Once we confirm, the checks will be created for us and will start running according to the schedule we have selected.
+
+Based on how comprehensively the file we imported described the API we want to monitor, we might have to tweak certain checks manually before they are fully set up.
+
+{{< figure src="/guides/images/guides-checkly-openapi-fail.png" alt="screenshot of checkly checks - with failures" >}}
+
+This usually includes adding missing headers or body contents that might not have been explicitly defined in the imported file.
+
+We can open the newly created checks and change any of their settings.
+
+{{< figure src="/guides/images/guides-checkly-openapi-body.png" alt="screenshot of checkly filled in check body" >}}
+
+After a small amount of tinkering, or none at all, we should have all our checks up and running.
+
+{{< figure src="/guides/images/guides-checkly-openapi-pass.png" alt="screenshot of checkly checks - all passing" >}}
+
+For APIs medium and large, and for any that is already described with an OAS file, this check-generating procedure can save us a large amount of time.
```

Comment:
Good article.
General comments:
1. external links need **_target=_blank_**
2. Can you be a bit more vocal about the benefits of creating checks via OpenAPI spec? Also should I create checks for every endpoint that is defined in the OpenAPI spec?

Reply:
1. Done. @tnolet maybe you know of a way to make this work with markdown? My little research says I either have to go with pure HTML (which I have done this time) or install some other markdown-like package.
2. Done.

Reply:
@ragog on 1) https://adityatelange.in/blog/hugo-external-link-new-tab/

Reply:
@hlenke good enough. I wish there was a pure markdown solution for this :) thanks
I am opening a new PR and converting all previous guides links.